core/domain/state_domain_test.py

```diff
@@ -64,43 +109,203 @@ def test_get_all_html_content_strings(self):
+                '<p>Equals rule_spec html</p>',
+                '<p>ContainsAtLeastOneOf rule_spec html</p>',
+                '<p>IsProperSubsetOf rule_spec html</p>',
+                '<p>DoesNotContainAtLeastOneOf rule_spec html</p>',
+                '',
+                '<p>init_state customization arg html 1</p>',
+                '<p>init_state customization arg html 2</p>',
+                '<p>init_state customization arg html 3</p>',
+                '<p>init_state customization arg html 4</p>',
+                '',
+                '<p>state2 written_translation content-hi</p>',
+                '<p>state2 written_translation content-en</p>',
+                '<p>state2 written_translation outcome-hi</p>',
+                '<p>state2 written_translation outcome-en</p>',
+                '<p>state2 written_translation feedback-hi</p>',
+                '<p>state2 written_translation feedback-en</p>',
+                '<p>State 2 Feedback</p>',
+                '<p>IsEqualToOrdering rule_spec htmls</p>',
+                '<p>HasElementXAtPositionY rule_spec html</p>',
+                '<p>y input for HasElementXAtPositionY rule_spec </p>',
+                '<p>x input for HasElementXAtPositionY rule_spec </p>',
+                ('<p>IsEqualToOrderingWithOneItemAtIncorrectPosition rule_s'
+                 'pec htmls</p>'), '',
+                '<p>state 2 customization arg html 1</p>',
+                '<p>state 2 customization arg html 2</p>',
+                '<p>state 2 customization arg html 3</p>',
+                '<p>state 2 customization arg html 4</p>',
+                '<p>state2 content html</p>',
+                '<p>state 1 outcome html</p>', '',
+                '<p>state 1 content html</p>'])
+
+        html_field_types_to_rule_specs_dict = json.loads(
+            utils.get_file_contents(
+                feconf.HTML_FIELD_TYPES_TO_RULE_SPECS_FILE_PATH))
+        for html_type_dict in (
+                html_field_types_to_rule_specs_dict.values()):
+            html_type_dict['format'] = 'invalid format'
+
+        def mock_get_file_contents(unused_file_path):
+            return json.dumps(html_field_types_to_rule_specs_dict)
+
+        with self.swap(utils, 'get_file_contents', mock_get_file_contents):
+            with self.assertRaisesRegexp(
+                Exception,
+                'The rule spec does not belong to a validformat.'):
+                html_list = (
+                    init_state.get_all_html_content_strings() +
+                    state2.get_all_html_content_strings() +
+                    state1.get_all_html_content_strings())
+        exploration_with_no_html = (
+            exp_domain.Exploration.create_default_exploration('exp_id2'))
+        init_state = (
+            exploration_with_no_html.states[
+                exploration_with_no_html.init_state_name])
+        html_list = init_state.get_all_html_content_strings()
+        self.assertEqual(html_list, ['', ''])
```

Comment:
I think you missed part of a previous comment about testing the "negative cases" with "Equals". As mentioned before, you should check that other cases with "Equals" are handled correctly, etc. Actively try to think about all places where things might go wrong and write tests for those.

Reply:
@seanlip, I have added the interaction `TextInput` Which has `Equals`.
Isn't that sufficient ?

Reply:
Ah, it's possible I didn't see that at all because it was in one giant test (I did look up and down it a couple of times). Yes, that would be sufficient -- sorry for missing that!

Reply:
done